mantle/kola/tests/misc/boot-mirror.go

```diff
@@ -0,0 +1,449 @@
+				  "devices": [
+					"/dev/disk/by-partlabel/boot-1",
+					"/dev/disk/by-partlabel/boot-2"
+				  ],
+				  "level": "raid1",
+				  "name": "md-boot",
+				  "options": [
+					"--metadata=1.0"
+				  ]
+				},
+				{
+				  "devices": [
+					"/dev/disk/by-partlabel/root-1",
+					"/dev/disk/by-partlabel/root-2"
+				  ],
+				  "level": "raid1",
+				  "name": "md-root"
+				}
+			  ]
+			}
+		  }`)
+)
+
+func init() {
+	register.RegisterTest(&register.Test{
+		Run:         runBootMirrorTest,
+		ClusterSize: 0,
+		Name:        `coreos.boot-mirror`,
+		Platforms:   []string{"qemu-unpriv"},
+		// skipping this test on UEFI until https://github.com/coreos/coreos-assembler/issues/2039
+		// gets resolved.
+		ExcludeFirmwares: []string{"uefi"},
+		Tags:             []string{"boot-mirror", "raid1"},
+		FailFast:         true,
+	})
+	register.RegisterTest(&register.Test{
+		Run:         runBootMirrorLUKSTest,
+		ClusterSize: 0,
+		Name:        `coreos.boot-mirror.luks`,
+		Platforms:   []string{"qemu-unpriv"},
+		// aarch64 and ppc64le are added temporarily to avoid test failure
+		// until we support specifying FCC directly in kola.
+		ExcludeArchitectures: []string{"aarch64", "ppc64le", "s390x"}, // no TPM support for s390x in qemu
+		// skipping this test on UEFI until https://github.com/coreos/coreos-assembler/issues/2039
+		// gets resolved.
+		ExcludeFirmwares: []string{"uefi"},
+		Tags:             []string{"boot-mirror", "luks", "raid1", "tpm2", kola.NeedsInternetTag},
+		FailFast:         true,
+	})
+}
+
+// runBootMirrorTest verifies if the boot-mirror RAID1
+// flow works properly in both BIOS and UEFI modes.
+func runBootMirrorTest(c cluster.TestCluster) {
+	var m platform.Machine
+	var err error
+	options := platform.QemuMachineOptions{
+		MachineOptions: platform.MachineOptions{
+			AdditionalDisks: []string{"5120M", "5120M"},
+			MinMemory:       4096,
+		},
+	}
+	m, err = c.Cluster.(*unprivqemu.Cluster).NewMachineWithQemuOptions(bootmirror, options)
+	if err != nil {
+		c.Fatal(err)
+	}
+	rootOutput := c.MustSSH(m, "sudo mdadm --export --detail /dev/md/md-root")
+	if !strings.Contains(string(rootOutput), "/dev/vda4") || !strings.Contains(string(rootOutput), "/dev/vdb4") || !strings.Contains(string(rootOutput), "/dev/vdc4") {
+		c.Fatalf("root raid device missing; expected devices: %v", string(rootOutput))
+	}
+	bootOutput := c.MustSSH(m, "sudo mdadm --export --detail /dev/md/md-boot")
+	if !strings.Contains(string(bootOutput), "/dev/vda3") || !strings.Contains(string(bootOutput), "/dev/vdb3") || !strings.Contains(string(bootOutput), "/dev/vdc3") {
+		c.Fatalf("boot raid device missing; expected devices: %v", string(bootOutput))
+	}
+	// Check for root
+	checkIfMountpointIsRaid(c, m, "/sysroot")
+	fsTypeForRoot := c.MustSSH(m, "findmnt -nvr /sysroot -o FSTYPE")
+	if strings.Compare(string(fsTypeForRoot), "xfs") != 0 {
+		c.Fatalf("didn't match fstype for root")
+	}
+	bootMirrorSanityTest(c, m)
+
+	detachPrimaryBlockDevice(c, m)
```

Comment:
We're currently trusting that the detach actually worked.  We could verify after reboot that the RAIDs are in degraded state; it shouldn't be too difficult.